src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -0,0 +1,223 @@
+        {
+            _nameResolver = nameResolver ?? new EmptyNameResolver();
+            _source = source;
+
+            Type t = typeof(TAttribute);
+
+            Dictionary<string, PropertyInfo> availableParams = new Dictionary<string, PropertyInfo>(StringComparer.OrdinalIgnoreCase);
+            foreach (var prop in t.GetProperties(BindingFlags.Instance | BindingFlags.Public))
+            {
+                var objValue = prop.GetValue(_source);
+                if (objValue != null)
+                {
+                    availableParams[prop.Name] = prop;
+                }
+            }
+
+            int longestMatch = -1;
+
+            // Pick the ctor with the longest parameter list where all parameters are matched. 
+            var ctors = t.GetConstructors(BindingFlags.Public | BindingFlags.Instance);
+            foreach (var ctor in ctors)
+            {
+                var ps = ctor.GetParameters();
+                int len = ps.Length;
+
+                var getArgFuncs = new Func<IReadOnlyDictionary<string, object>, object>[len];
+
+                bool ok = true;
+                for (int i = 0; i < len; i++)
+                {
+                    var p = ps[i];
+                    PropertyInfo propInfo = null;
+                    if (!availableParams.TryGetValue(p.Name, out propInfo))
+                    {
+                        ok = false;
+                        break;
+                    }
+                    bool resolve = propInfo.GetCustomAttribute<ResolvePropertyAttribute>() != null;
+                    getArgFuncs[i] = (bindingData) =>
+                    {
+                        var propValue = propInfo.GetValue(_source);
+                        if (resolve)
+                        {
+                            string str = (string)propValue;
+                            if (str != null)
+                            {
+                                str = _nameResolver.ResolveWholeString(str);
+
+                                BindingTemplate template = BindingTemplate.FromString(str);
+                                propValue = template.Bind(bindingData);
+                            }
+                        }
+                        return propValue;
+                    };
+                }
+
+                if (ok)
+                {
+                    if (len > longestMatch)
+                    {
+                        _bestCtor = ctor;
+                        longestMatch = len;
+                        _bestCtorArgBuilder = getArgFuncs;
+                    }
+                }
+            }
+
+            if (_bestCtor == null)
+            {
+                // error!!!
+                throw new InvalidOperationException("Can't figure out which ctor to call.");
+            }
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
```

Comment:
From the interface level, I'd like it to be treated as an instance method. 

---
In reply to: [60985743](https://github.com/Azure/azure-webjobs-sdk/pull/694#discussion_r60985743) [](ancestors = 60985743)